### ISSUE TITLE
IE11 fromJS bug 

### ISSUE THREAD

**Issue by rytisbac** [immutable-js/immutable-js#1648]
Version v4.0.0-rc.12 
Function fromJS. 
example:
`console.log(fromJS({test: 'qqq'}))`
In most of the browsers it returns Map object, but IE11 returns the same object. 
Other case:
`console.log(fromJS([{test: 'qqq'}]))`
In most of the browsers it returns List and Map as first element, but IE11 returns List and initial object. 

Version 3.8.2 works as expected

**Comment by thenano** [comment]
Any chance this will be addressed? It completely broke my application on internet explorer as I use fromJS a lot.
I can confirm that the issue was introduced with 4.0.0-rc10, up to  4.0.0-rc9 everything works fine

**Comment by cauethenorio** [comment]
@thenano I had the same issue and was able to fix it installing the `core-js` `es6.function.name` polyfill.

**Comment by yuya-sato-geniee** [comment]
List structure seems to be broken by using .concat() on IE11 on 4.0.0-rc.12.

But v3.8.2 works good.

Is it related to this issue?

### How to reproduce

Please open below by IE11.

[http://another.geniee.jp/test/ysato/immutable-bug/map.html](http://another.geniee.jp/test/ysato/immutable-bug/map.html)

```javascript
var MyRecord = Immutable.Record({a: 1});
var list = Immutable.List([]);
list = list.concat([new MyRecord({a: 2}), new MyRecord({a: 3})]);
console.log(JSON.stringify(list)); 
// from Chrome: [{"a":2},{"a":3}]
// from IE11: [[{"a":2},{"a":3}]]     <- it is incorrect
```

**Comment by sjfkai** [comment]
I had the same issue with `fromJS`

**Comment by hiyangguo** [comment]
Me too. And when I execute the code :
```
Immutable.fromJS({a:1}).toJS()
```
In IE 11 ，it will throw exception `Object don't support "toJS" prop or function`，but in chrome it is work ok.
#### package.json
```
"immutable": "^4.0.0-rc.12",
```

**Comment by hiyangguo** [comment]
But when I use the 
```
"immutable": "^3.8.1"
```
It is work ok.

**Comment by alexmassicott** [comment]
@cauethenorio You're the real mvp, the polyfill fixed it for me.

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/f3a6d5ce75bb9d60b87074240838f5429e896b60
